components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -226,6 +226,7 @@ open class GleanInternalAPI internal constructor () {
             val packageInfo = applicationContext.packageManager.getPackageInfo(
                 applicationContext.packageName, 0
             )
+            @Suppress("DEPRECATION")
```

Comment:
Mh, woha, `versionCode` is being deprecated? I wonder if there's anything we should do here before we start relying on this data.

Reply:
Yes -- but unfortunately, its replacement (`getLongVersionCode()`) isn't available in the earliest version we need to support, so it fails in testing.

Reply:
Ouch, I see :) Suppressing is fine then :)

Reply:
There are ways of doing version dependent code branches, is that something worth looking at doing here?

Reply:
The whole thing is a bit weird.  It seems like they changed the version identifier from an `int` to a `long`, with the latter having what's called a "major version id" in the upper 32 bits.  We don't appear use or care about that, so we'd be masking those bits out.  All that seemed much more brittle to me than just using the deprecated API (that's sure to do the right thing) as long as it exists.

Reply:
That's perfectly reasonable to me, I doubt they ever do more than deprecate it.